junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -314,19 +526,31 @@ fun <R> assertTimeoutPreemptively(
  * }
  * ```
  * @see Assertions.assertTimeoutPreemptively
- * @paramR the result of the [executable].
+ * @param R the result of the [executable].
  */
+@OptIn(ExperimentalContracts::class)
 @API(status = STABLE, since = "5.11")
 fun <R> assertTimeoutPreemptively(
     timeout: Duration,
     message: () -> String,
     executable: () -> R
-): R = Assertions.assertTimeoutPreemptively(timeout, executable, message)
+): R {
+    contract {
+        callsInPlace(message, AT_MOST_ONCE)
```

Comment:
Shouldn't executable also have a contract?

Reply:
Good catch! Added in #4192.

Reply:
It was ignored explicitly. Please, see [this comment](https://github.com/junit-team/junit5/pull/3259#discussion_r1315191360).

Reply:
Oh, I had forgotten that discussion. Thanks for the reminder! :+1:

Reply:
Now I wonder, whether we should add the explanation why some of the lambdas don't have a contract. I'm not sure, if the actual code is the best place, though.

Reply:
Could be a single sentence in code (each method), and a section at the top/bottom of the file?

Reply:
I think a single comment in each of those methods should suffice. Something like
```
// No contract for `executable` because ...
```

Reply:
Most likely I won't be able to do this in December.
I may take a look in January,  if someone else hasn't done that by then.

Reply:
I've taken care of it in ebf60bb8672367b07de9e3750269431dd5878c82